src/NLog/Targets/DatabaseTarget.cs

```diff
@@ -657,7 +692,7 @@ private void WriteLogEventBatchToDatabase(IList<AsyncLogEventInfo> logEvents, st
                 //Always suppress transaction so that the caller does not rollback logging if they are rolling back their transaction.
                 using (TransactionScope transactionScope = new TransactionScope(TransactionScopeOption.Suppress))
                 {
-                    EnsureConnectionOpen(connectionString);
+                    EnsureConnectionOpen(connectionString, logEvents.Count > 0 ? logEvents[0].LogEvent : null);
```

Comment:
There was a reason for not using Linq, isn't?

Reply:
So many traps in Linq, so I try to avoid it, unless writing unit-testing code. Then allocation is free.

Reply:
Could we add our own .FirstOrDefault (now or in the future)? It;s some much cleaner

Reply:
Feel free